contracts/layer-2/ChampMarketplace.sol

```diff
@@ -258,10 +252,15 @@ contract ChampMarketplace is AccessControlEnumerableUpgradeable {
             tokenWeiPrice > 0,
             "ChampMarketplace: Price should be strictly positive"
         );
+        require(
+            nftOwner != reserve,
+            "ChampMarketplace: Can not reserve sale for token owner."
+        );
 
         _sales[tokenId] = tokenWeiPrice;
+        _reservedOffers[tokenId] = reserve;
```

Comment:
Write to storage without regard for any existing value to maintain consistency with the price update pattern (line 260)